recipes/symengine/all/conanfile.py

```diff
@@ -0,0 +1,91 @@
+from conans import ConanFile, CMake, tools
+import os
+
+required_conan_version = ">=1.33.0"
+
+
+class SymengineConan(ConanFile):
+    name = "symengine"
+    description = "A fast symbolic manipulation library, written in C++"
+    license = "MIT"
+    topics = ("symbolic", "algebra")
+    homepage = "https://symengine.org/"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "integer_class": ["boostmp", "gmp"],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "integer_class": "boostmp",
+    }
+    short_paths = True
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def requirements(self):
+        if self.options.integer_class == "boostmp":
+            self.requires("boost/1.76.0")
+        else:
+            self.requires("gmp/6.2.1")
+
+    def source(self):
+        tools.get(
+            **self.conan_data["sources"][self.version],
+            strip_root=True,
+            destination=self._source_subfolder,
+        )
+
+    def _configure_cmake(self):
+        if self._cmake is None:
+            self._cmake = CMake(self)
+            self._cmake.definitions["BUILD_TESTS"] = False
+            self._cmake.definitions["BUILD_BENCHMARKS"] = False
+            self._cmake.definitions["INTEGER_CLASS"] = self.options.integer_class
+            self._cmake.definitions["MSVC_USE_MT"] = False
+            self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        # [CMAKE-MODULES-CONFIG-FILES (KB-H016)]
+        tools.remove_files_by_mask(self.package_folder, "*.cmake")
+        # [DEFAULT PACKAGE LAYOUT (KB-H013)]
+        tools.rmdir(os.path.join(self.package_folder, "CMake"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["symengine"]
+        if self.settings.build_type == "Debug":
+            self.cpp_info.libs.append("teuchos")
```

Comment:
Looks like it's more complicated.
There is a `WITH_SYMENGINE _TEUCHOS` option: https://github.com/symengine/symengine/blob/301b7a63584981a4da193a5163bf606057661e86/CMakeLists.txt#L531-L532

But it's conditionally force enabled: https://github.com/symengine/symengine/blob/301b7a63584981a4da193a5163bf606057661e86/CMakeLists.txt#L571-L573

```suggestion
        self.cpp_info.libs = sorted(tools.collect_libs(self), key = lamda v: ["teuchos", "symengine"].index(v))
```

Reply:
Has to be the other order I think. Done. Fingers crossed this time.

Reply:
Indeed, other order.
You weren't sleeping :wink: